server/src/main/java/org/opensearch/search/profile/query/ConcurrentQueryProfileBreakdown.java

```diff
@@ -23,7 +28,21 @@
  * @opensearch.internal
  */
 public final class ConcurrentQueryProfileBreakdown extends ContextualProfileBreakdown<QueryTimingType> {
-    private final Map<Object, AbstractProfileBreakdown<QueryTimingType>> contexts = new ConcurrentHashMap<>();
+    static final String SLICE_END_TIME_SUFFIX = "_slice_end_time";
+    static final String SLICE_START_TIME_SUFFIX = "_slice_start_time";
+    static final String MAX_PREFIX = "max_";
+    static final String MIN_PREFIX = "min_";
+    static final String AVG_PREFIX = "avg_";
+    private long queryNodeTime = Long.MIN_VALUE;
+    private long maxSliceNodeTime = Long.MIN_VALUE;
+    private long minSliceNodeTime = Long.MAX_VALUE;
+    private long avgSliceNodeTime = 0L;
+
+    // keep track of all breakdown timings per segment. package-private for testing
+    final Map<Object, AbstractProfileBreakdown<QueryTimingType>> contexts = new ConcurrentHashMap<>();
+
+    // represents slice to leaves mapping as for each slice a unique collector instance is created
+    private final Map<Collector, List<LeafReaderContext>> sliceCollectorToLeaves = new HashMap<>();
```

Comment:
I believe the profile is shared between index searcher threads, right?

```suggestion
    private final Map<Collector, List<LeafReaderContext>> sliceCollectorsToLeaves = new ConcurrentHashMap<>();
```